DownFall/Views/HUD.swift

```diff
@@ -71,50 +68,73 @@ class HUD: SKSpriteNode {
+                                                 relativeTo: goldLabel.frame,
+                                                 verticaliy: .center,
+                                                 anchor: .right)
         self.addChild(coinNode)
         
-        if let _ = self.childNode(withName: "goldLabel") {
-            
-        }
-        else {
-            let goldLabel = SKLabelNode(fontNamed: "Helvetica")
-            goldLabel.fontSize = 75
-            goldLabel.position = CGPoint(x: 355, y: -22)
-            goldLabel.text = "0"
-            goldLabel.fontColor = .lightText
-            goldLabel.name = "goldLabel"
-            self.addChild(goldLabel)
-        }
-    }
-    
-    func showGem() {
-        let spriteNode = SKSpriteNode(texture: SKTexture(imageNamed: "gem1"), size: CGSize(width: 100, height: 100))
-        spriteNode.position = CGPoint(x: -300, y: 0)
-        self.addChild(spriteNode)
+        let gemSpriteLabel = ParagraphNode(text: "\(data.carry.total(in: .gem))", paragraphWidth: Style.HUD.labelParagraphWidth, fontName: UIFont.pixelFontName, fontSize: UIFont.extraLargeSize, fontColor: .lightText)
+        gemSpriteLabel.position = CGPoint.positionThis(gemSpriteLabel.frame,
+                                                       relativeTo: coinNode.frame,
+                                                       verticaliy: .center,
+                                                       anchor: .right,
+                                                       xOffset: Style.HUD.coinLabelPadding)
+        gemSpriteLabel.name = Identifiers.gemSpriteLabel
+        self.addChild(gemSpriteLabel)
+        
+        let gemSprite = SKSpriteNode(texture: SKTexture(imageNamed: Identifiers.gem), size: Style.HUD.gemSize)
+        gemSprite.position = CGPoint.positionThis(gemSprite.frame, relativeTo: gemSpriteLabel.frame, verticaliy: .center, anchor: .right, xOffset: Style.HUD.gemSpritePadding)
+        self.addChild(gemSprite)
+        
+        
+        
     }
     
-    func incrementGoldCounter() {
-        if let goldLabel = self.childNode(withName: "goldLabel") as? SKLabelNode,
-            let text = goldLabel.text,
-            let gold = Int(text) {
-            goldLabel.text = ""
-            goldLabel.text = "\(gold + 1)"
+    func incrementCurrencyCounter(_ item: Item, total: Int) {
+        if item.type == .gold,
+            let goldLabel = self.childNode(withName: Identifiers.goldSpriteLabel) as? ParagraphNode {
+            let oldPosition = goldLabel.position
+            goldLabel.removeFromParent()
+            
+            let newGoldLabel = ParagraphNode(text: "\(total)", paragraphWidth: Style.HUD.labelParagraphWidth, fontName: UIFont.pixelFontName, fontSize: UIFont.extraLargeSize, fontColor: .lightText)
+            newGoldLabel.position = oldPosition
+            newGoldLabel.name = Identifiers.goldSpriteLabel
+            self.addChild(newGoldLabel)
```

Comment:
This code is duplicated.

Reply:
Fixed.  I cannot update the earlier code in the setup func because it requires to know about other sprites for positioning logic.